### ISSUE TITLE
Comments on aurelia-realworld-example-app

### ISSUE THREAD

**Issue by adriatic** [aurelia-community/community-wide-collaboration#14]
The app lives here https://github.com/jadhavajay/aurelia-realworld-example-app and was offered as a contribution to AUCS project. While the app looks interesting based in its description (**[README.md](https://github.com/jadhavajay/aurelia-realworld-example-app/blob/master/README.md)**) I failed to build it on my machine. **`npm install`** creates too many **`gyp`** errors to list here, so I used **`yarn`**  instead. The number of errors was smaller and I believe that they are caused by not finding **`python 2`**. 

I would like to add this app to the collection of ""real world aurelia full stack apps" (project that is also on the AUCS task list, just not defined yet) and that the detailed description of its architecture and code becomes a blog. Since this app covers several topics that we will address in form of guides, there should be references from this blog article into the guides - and vice versa.

**Comment by smoore4moma** [comment]
This is a task I have to do soon anyway, but it would be good to have a model .gitignore for an Aurelia CLI generated app posted to GitHub.  The one that is generated with au new looks good but I wonder if there are other things that should be excluded.  

Also, what is the best way to handle say an api key for a plug-in in main.js (Google maps and analytics for example)?  Do I need a Jenkins or Bamboo for that?  Noon question, I know, but I am not sure.

**Comment by fmonroe** [comment]
OK - I give up - what's a "noon question" ? Is it the new, fancier, hip, term-of-the-day for "noob"? 😀

**Comment by adriatic** [comment]
@smoore4moma the Aurelia CLI Guide will soon define all details you are presenting **[above](https://github.com/aurelia-community/community-wide-collaboration/issues/14#issuecomment-360817284)** - and hopefully more. You may check the old version of this document **[here](https://aurelia-tools.gitbooks.io/aurelia-cli-visions/content/)**

As I am still bouncing back and forth defining the **[AUCS Development Environment](https://github.com/aucs-publishers/AUCS-guide-gitbook/issues/1)**, which is the key missing part for AUCS succesfull start and subsequent life, I am not ready to define all details which should like everything else be discussed before being defined firmly

**Comment by smoore4moma** [comment]
Ha!  no i meant noob @fmonroe.

**Comment by smoore4moma** [comment]
Perhaps create a global Aurelia.gitignore here?:  https://github.com/github/gitignore/tree/master/Global

**Comment by loaded02** [comment]
@adriatic These errors sound as they are related to node-sass. I had similar issues. This is because I am using sass in aurelia-cli setup which depends on node-sass. There is an issue about node-sass and npm [here](https://github.com/sass/node-sass/issues/2101). I solved it on a Windows PC with running ``` npm install --global --production windows-build-tools``` in admin mode. This will install python and compile node-sass for win x64 architecture. It's not an issue on Linux. I checked out the project today and it is still working like it should. I can't reproduce your errors now. Keep me posted if this solves your issue. Otherwise I could get rid of sass to make the process easier for other users.

**Comment by adriatic** [comment]
@smoore4moma Good idea (please keep it alive in your memory as I am running out of space where to kepp all of my todos 😏 ). Note also that we will have a very different .gitignore for repos with MD documents from the repos with guides related sample code.

**Comment by loaded02** [comment]
@adriatic Try this [branch](https://github.com/loaded02/aurelia-realworld-example-app/tree/no-sass) without sass and with yarn.

**Comment by adriatic** [comment]
@loaded02 thanks for the additional info on your sample, regardless of my plan not to dive any deeper into it now, before we get AUCS project going. So my **[comments](https://github.com/aurelia-community/community-wide-collaboration/issues/14#issue-291932195)** were written as a personal note, to be used when we get back to it.

**Comment by smoore4moma** [comment]
I know the goal is the skeleton navigation examples, but the ToDo tutorial
also need work.  I thought it would be 10 minutes to get it running (wanted
to do it before starting an intern on the same).  Worked out the issues and
posted working esnext version:

https://github.com/smoore4moma/aurelia-todo-example

Steve

Steven MOORE

Database Administrator

Department of Information Technology


The Museum of Modern Art

11 West 53 Street

New York, NY 10019
T (813) 240 0587

**Comment by adriatic** [comment]
Good to see that this issue is still alive as I care a lot about application authentication / authorization (aka identity management). The goal for AUCS guides / samples is a lot wider than just "coverage of skeletons", so I am quite interested in seeing how dis you solve this one.

Just for the full synchronization between us: the plan is to 
1. Create the navigation skeletons "based" on Aurelia CLI (the original skeletons are still SystemJS / JSPM based). These new skeletons are finished and accessible **[here](https://github.com/longarone/code-samples/tree/master/basic)**
2. Augments these **[cli skeletons](https://github.com/longarone/code-samples/tree/master/basic)** by adding the support for several (Auth0, Amazon, ...) Identity Management Platform as a Service (PaaS) solution providers.
3... more to come ...
